Library/Homebrew/extend/os/linux/system_config.rb

```diff
@@ -40,10 +40,20 @@ def host_ruby_version
       out
     end
 
+    def wsl_version(kernel)
+      return "no" unless /-microsoft/i.match?(kernel)
+
+      return "2 (store)" if Version.new kernel[/Linux ([0-9.]*)-.*/, 1] > Version.new("5.15")
+      return "2" if /-microsoft/.match?(kernel)
+      return "1" if /-Microsoft/.match?(kernel)
+    end
+
     def dump_verbose_config(out = $stdout)
+      kernel = Utils.safe_popen_read("uname", "-mors").chomp
       dump_generic_verbose_config(out)
-      out.puts "Kernel: #{`uname -mors`.chomp}"
+      out.puts "Kernel: #{kernel}"
       out.puts "OS: #{OS::Linux.os_version}"
+      out.puts "WSL: #{wsl_version(kernel)}"
```

Comment:
```suggestion
      if (wsl = wsl_version(kernel).presence)
        out.puts "WSL: #{wsl}"
      end
```

Let's not print anything here on the typical, non-WSL case.